src/heretic/sb_bar.c

```diff
@@ -1051,69 +1051,169 @@ void DrawInventoryBar(void)
 
 void DrawFullScreenStuff(void)
 {
-    const char *patch;
+	const char *patch;
     int i;
     int x;
     int temp;
 
     UpdateState |= I_FULLSCRN;
-    if (CPlayer->mo->health > 0)
```

Comment:
Could you please try to create the smallest possible diff, i.e. preserve as much original code as possible (including its indentation), and try to mostly add me code?

Reply:
I tried but failed. Do you have any suggestions how to make it a cleaner diff in this example? I think a conditional check is needed, alternatively a DrawCrispyFullScreenStuff could be introduced and the check is moved to SB_Drawer.

Reply:
I'd be fine if you could just leave the original indentation intact, even if this would introduce inconsistencies.

As a rule of thumb, I try to avoid changing this code

```c
if (old_condition)
{
    old_code();
}
```

into this

```c
if (old_condition or new_condition)
{
    if (old_condition)
    {
        old_code();
    }
    else if (new_condition)
    {
        new_code();
    }
}
```

but into this instead

```c
if (new_condition)
{
    new_code();
}
else
if (old_condition)
{
    old_code();
}
```

Reply:
Thanks for the input! 
I did an update to this approach first, is this also ok from your side?

```c
if (new_condition)
{
    new_code();
    return;
}
old_code();
```

Otherwise, I'll change it to go with the inconsistencies.